nevergrad/optimization/optimizerlib.py

```diff
@@ -762,6 +763,94 @@ def _internal_provide_recommendation(self) -> ArrayLike:
         return self.avg
 
 
+@registry.register
+class Splitter(base.Optimizer):
+    """Combines optimizers, each of them working on their own variables.
+    
+    num_optims: number of optimizers
+    num_vars: number of variable per optimizer.
+
+    E.g. for 5 optimizers, each of them working on 2 variables, we can use:
+    opt = Splitter(instrumentation=10, num_workers=3, num_optims=5, num_vars=[2, 2, 2, 2, 2])
+    or equivalently:
+    opt = Splitter(instrumentation=10, num_workers=3, num_vars=[2, 2, 2, 2, 2])
+    Given that all optimizers have the same number of variables, we can also do:
+    opt = Splitter(instrumentation=10, num_workers=3, num_optims=5)
+
+    This is 5 parallel (by num_workers = 5).
+
+    Be careful! The variables refer to the deep representation used by optimizers.
+    For example, a categorical variable with 5 possible values becomes 5 continuous variables.
+    """
+
+    def __init__(self, instrumentation: Union[int, Instrumentation], budget: Optional[int] = None, num_workers: int = 1, num_optims: Optional[int] = None, num_vars: Optional[List[Any]] = None) -> None:
+        super().__init__(instrumentation, budget=budget, num_workers=num_workers)
+        if num_vars:
+            if num_optims:
+                assert num_optims == len(num_vars)
+            else:
+                num_optims = len(num_vars)
+            assert sum(num_vars) == self.dimension
+        else:
+            if not num_optims:  # if no num_vars and no num_optims, just assume 3.
+                num_optims = 2
+            # num_vars not given: we will distribute variables equally.
+        if num_optims > self.dimension:
+            num_optims = self.dimension
+            if num_vars:
+                num_vars = num_vars[:num_optims]
+        self.num_optims = num_optims
+        self.optims: List[Any] = []
+        self.num_vars: List[Any] = num_vars if num_vars else []
+        self.instrumentations: List[Any] = []
+        self.num_yoyo = 0
+        for i in range(self.num_optims):
+            if not self.num_vars or len(self.num_vars) < i+1:
+                self.num_vars += [(self.dimension // self.num_optims) + (self.dimension % self.num_optims > i)]
+            
+            assert len(self.num_vars) > 0
+            assert self.num_vars[-1] >= 0
```

Comment:
I'm not sure how informative this assert is.
It would feel cleaner to check that with something like all(var_dim > 0 for var_dim in num_vars) at the end of that loop.

Reply:
added a check for all i, thx